arb_os/gasAccounting.mini

```diff
@@ -9,62 +9,220 @@ use std::bytearray::MarshalledBytes;
+                maxRate1: fixedPoint_new(const::NetFee_maxRate1Num, const::NetFee_maxRate1Denom),
+                maxRate2: fixedPoint_new(const::NetFee_maxRate2Num, const::NetFee_maxRate2Denom),
+            }
+        );
+    }
+
     gasAccountingInfo = struct {
-        gasUsedByOS: 0,
-        weiCollectedFromTxs: 0,
+        gasPool: gasAccountingParams.GasPoolMax,
+        reserve: 0,
+        owedToValidators: 0,
+        totalPaidToValidators: 0,
+        specialDebts: specialDebts_new(),
+        currentGasPrice: 0,
+        gasRemainingThisBlock: 0,
+        lastBlockNumSeen: 0,
         txState: None<TxGasState>,
+        pricingState: struct {
+            costRecoveryPrice: 0,
+            previousAmountOwed: 0,
+            congestionPrice: 0,
+        },
+        payValFeesTo: None<address>,
+        netFeeInfo: netFeeInfo,
+        overheadTracker: overheadTracker_new(5, 0),
+    };
+}
+
+public impure func gasAccounting_setFeeParams(baseGasPrice: uint, storageCharge: uint, validatorFeeRecipient: address) {
+    gasAccountingParams = gasAccountingParams with {
+        BaseGasPrice: baseGasPrice
+    } with {
+        StorageCharge: storageCharge
+    };
+    let recipient = None<address>;
+    if (validatorFeeRecipient != address(0)) {
+        recipient = Some(validatorFeeRecipient);
+    }
+
+    gasAccountingInfo = gasAccountingInfo with {
+        payValFeesTo: recipient
+    } with {
+        currentGasPrice: baseGasPrice
     };
 }
 
+public impure func gasAccounting_postInitMessage() {
+    let millisecondsPerBlock = chainParams_millisecondsPerBlock();
+    let speedLimitPerSecond = chainParams_speedLimitPerSecond();
+    let speedLimitPerBlock = millisecondsPerBlock * speedLimitPerSecond / 1000;
+    gasAccountingParams = struct {
+        SpeedLimitPerBlock: speedLimitPerBlock,
+        GasPoolMax: int(5 * speedLimitPerBlock),
+        BaseGasPrice: 0,
+        TxGasLimit: 2 * speedLimitPerBlock,
+        StorageCharge: 0,
+    };
+    gasAccountingInfo = gasAccountingInfo with { gasPool: gasAccountingParams.GasPoolMax };
+}
+
+public impure func gasAccounting_summaryToPublish() -> any {
+    let payoutAddress = address(0);
+    if let Some(paddr) = gasAccountingInfo.payValFeesTo {
+        payoutAddress = paddr;
+    }
+
+    return (
+        gasAccountingInfo.currentGasPrice,
+        gasAccountingInfo.gasPool,
+        int(gasAccountingInfo.reserve) - int(gasAccountingInfo.owedToValidators),
+        gasAccountingInfo.totalPaidToValidators,
+        payoutAddress,
+    );
+}
+
+public impure func gasAccounting_addToReserveFunds(amount: uint) {
+    gasAccountingInfo = gasAccountingInfo with {
+        reserve: gasAccountingInfo.reserve + amount
+    };
+}
+
+impure func addDelayedOverhead(amount: uint) {
+    gasAccountingInfo = gasAccountingInfo with {
+        overheadTracker: overheadTracker_add(gasAccountingInfo.overheadTracker, amount)
+    };
+}
+
+public impure func recycleBurnedFunds(amount: uint) {
```

Comment:
This appears to be currently unused

Reply:
Keeping this around because it's likely to be needed later.